src/sentry/api/endpoints/user_notification_details.py

```diff
@@ -24,20 +56,44 @@ def get_attrs(self, item_list, user, *args, **kwargs):
         return results
 
     def serialize(self, obj, attrs, user, *args, **kwargs):
-        data = {option.key: option.value for option in attrs}
 
-        return {
-            'alertEmail': data.get('alert_email'),
-            'deployEmails': data.get('deploy-emails'),
-            'seenReleaseBroadcast': data.get('seen_release_broadcast'),
-            'selfAssignIssue': data.get('self_assign_issue'),
-            'selfNotifications': data.get('self_notifications'),
-            'subscribeByDefault': data.get('subscribe_by_default'),
-            'workflowNotifications': data.get('workflow:notifications'),
-        }
+        raw_data = {option.key: option.value for option in attrs}
+
+        data = {}
+        for key in USER_OPTION_SETTINGS:
+            uo = USER_OPTION_SETTINGS[key]
+            val = raw_data.get(uo['key'], uo['default'])
+            if (uo['type'] == bool):
+                data[key] = bool(int(val))  # '1' is true, '0' is false
+            elif (uo['type'] == int):
+                data[key] = int(val)
```

Comment:
slightly worried me that things get silently dropped if their type isn't bool or int, but that paranoia might not be worth adding a raise or a log statement